cmd/nerdctl/logs.go

```diff
@@ -129,47 +130,25 @@ func logsAction(cmd *cobra.Command, args []string) error {
 				if err != nil {
 					return err
 				}
-				var reader io.Reader
-				var execCmd *exec.Cmd
-				//chan for non-follow tail to check the logsEOF
-				logsEOFChan := make(chan struct{})
-				if follow && status.Status == containerd.Running {
-					waitCh, err := task.Wait(ctx)
-					if err != nil {
-						return err
-					}
-					reader, execCmd, err = newTailReader(ctx, task, logJSONFilePath, follow, tail)
-					if err != nil {
-						return err
-					}
-
-					go func() {
-						<-waitCh
-						execCmd.Process.Kill()
-					}()
-				} else {
-					if tail != "" {
-						reader, execCmd, err = newTailReader(ctx, task, logJSONFilePath, false, tail)
-						if err != nil {
-							return err
-						}
-						go func() {
-							<-logsEOFChan
-							execCmd.Process.Kill()
-						}()
+				if status.Status != containerd.Running {
+					follow = false
+				}
 
-					} else {
-						f, err := os.Open(logJSONFilePath)
-						if err != nil {
-							return err
+				// channel for read log EOF
+				logsEOFChan := make(chan struct{})
+				reader, execCmd, err := newTailReader(ctx, logJSONFilePath, follow, tail)
+				if err != nil {
+					return err
+				}
+				go func() {
+					if follow {
+						if waitCh, err := task.Wait(ctx); err == nil {
+							<-waitCh
 						}
-						defer f.Close()
-						reader = f
-						go func() {
-							<-logsEOFChan
-						}()
+						execCmd.Process.Kill()
 					}
-				}
+					<-logsEOFChan
```

Comment:
Why ` <- logsEOFChan`

Reply:
Nice question!

`logsEOFChan` is an unbuffered channel, the receiver needs to be ready, [sender](https://github.com/containerd/nerdctl/blob/master/pkg/logging/jsonfile/jsonfile.go#L87) can be sent.

But seems like the channel does not be used, I will remove it.